File: backend/src/main/java/com/blackopalsolutions/goneviral/backend/model/response/GetRoleResponse.java
package com.blackopalsolutions.goneviral.backend.model.response;

import com.blackopalsolutions.goneviral.backend.model.domain.Role;

public class GetRoleResponse extends Response {
    private final Role role;

    public GetRoleResponse(Role role) {
        super(true);
        this.role = role;
    }

    public GetRoleResponse(String message) {
        super(message);
        role = null;
    }

    public Role getRole() {
        return role;
    }
}

File: shared/src/commonMain/kotlin/com/blackopalsolutions/goneviral/model/response/GetRoleResponse.kt
package com.blackopalsolutions.goneviral.model.response

import com.blackopalsolutions.goneviral.model.domain.Role

class GetRoleResponse : Response {
    val role: Role?

    constructor(role: Role) : super(true) {
        this.role = role
    }

    constructor(message: String) : super(message) {
        role = null
    }
}